optuna/storages/_rdb/storage.py

```diff
@@ -769,7 +769,7 @@ def _check_and_set_param_distribution(
             # Acquire lock.
             #
             # Assume that study exists.
-            models.StudyModel.find_by_id(study_id, session, for_update=True)
+            models.StudyModel.find_or_raise_by_id(study_id, session, for_update=True)
```

Comment:
As mentioned in the comment above, the implementation should be able to assert that the study exists. Sorry but could you elaborate a bit on your motivation, for instance if you depended on this change somewhere else?

Reply:
The main motivation for this change is to reduce the number of methods in `models.py` at #2062. This is the only use of `StudyModel.find_by_id`. Is it impossible to remove it?

Reply:
Okay, sounds reasonable. Thanks for clarifying that.